topy/topy.py

```diff
@@ -121,9 +121,22 @@ def sanitize_filename(filename):
 def print_diff(filename, old, new, stream=sys.stdout):
     """Diffs the `old` and `new` strings and prints as unified diff to file-like object `stream`."""
 
-    # TODO: color output for terminals
     lines = unified_diff(old.splitlines(True), new.splitlines(True), filename, filename)
-    stream.writelines(lines)
+    lines_color = add_output_color(lines)
+    stream.writelines(lines_color if not None else lines)  # print in colour if available
+
+
+def add_output_color(lines):
+    """Adds color to the output of the diff lines."""
+    try:
+        lines_list = list(lines)
```

Comment:
Note that `unified_diff()` returns an iterator, which yields lines one by one. This means the whole diff doesn't have to be loaded into memory at once, and can help with processing large files.

So instead of forcing the whole diff into a list, better use `map()` with a callback. map also returns an iterator that handles lines one by one.

Reply:
Cool, great advice - thanks. I'll implement it with a map